dist/privacy.js

```diff
@@ -291,10 +315,31 @@ return JSON.stringify({
    */
   geolocation: {
     navigator_geolocation_getCurrentPosition: testPropertyStringInResponseBodies(
-      'navigator.+geolocation.+getCurrentPosition'
+      'geolocation.+getCurrentPosition'
     ),
     navigator_geolocation_watchPosition: testPropertyStringInResponseBodies(
-      'navigator.+geolocation.+watchPosition'
+      'geolocation.+watchPosition'
     ),
-  }
+  },
+
+  /**
+   * List of hostnames with CNAME record
+   */
+  request_hostnames_with_cname: (() => {
+    let results = {};
+
+    for (const request of $WPT_REQUESTS) {
+      request_hostname = (new URL(request.url)).hostname;
+
+      for (const [origin, dns_info] of Object.entries($WPT_DNS)) {
+        dns_hostname = (new URL(origin)).hostname;
```

Comment:
@max-ostapenko I'm obviously late to reviewing this, but `new URL` can throw on an invalid URL, which would cause the entire metric recording to fail since there's no error handling. This should probably be wrapped in a try/catch.

(I have no examples of this actually occurring)

Reply:
Thanks

Reply:
Oh btw I fixed this in #130